sql/core/src/test/resources/sql-tests/inputs/datetime.sql

```diff
@@ -122,3 +122,13 @@ select to_timestamp("2019-10-06T10:11:12'12", "yyyy-MM-dd'T'HH:mm:ss''SSSS"); --
 select to_timestamp("2019-10-06T10:11:12'", "yyyy-MM-dd'T'HH:mm:ss''"); -- tail
 select to_timestamp("'2019-10-06T10:11:12", "''yyyy-MM-dd'T'HH:mm:ss"); -- head
 select to_timestamp("P2019-10-06T10:11:12", "'P'yyyy-MM-dd'T'HH:mm:ss"); -- head but as single quote
+
+-- Unsupported narrow text style
+select date_format(date '2020-05-23', 'GGGGG');
```

Comment:
shall we also fail for the parsing functions?

Reply:
This question may beyond the scope of this PR. So far, using an invalid pattern for parsing always returns `NULL`, which is also compatible with 2.4.

Reply:
I just realize that returning NULL is also silent data change...

Reply:
For parsing and formatting, throw SparkUpgradeExeception to tell users how to restore or update their pattern via our online doc? How about the current approach